### PR TITLE
CMR-7536 MimeType bulk granule update functionality for ECHO10 granules

### PR DESCRIPTION
**Pull request by jaybarra**
Allows the ECHO10 OnlineResources and OnlineAccessURLs to have the associated MimeType values updated.

### CODE REVIEW

File: ingest-app/src/cmr/ingest/services/granule_bulk_update/mime_type/echo10.clj [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+
+(defn update-mime-type
+  "Update the the MimeType for elements within OnlineResources and OnlineAccess in echo10
+   granule metadata and return granule."
+  [concept links]
+  (let [parsed (xml/parse-str (:metadata concept))
+        url-map (apply merge (map #(hash-map (:URL %) (:MimeType %)) links))
+        _ (when-not (= (count (set (keys url-map))) (count links))
+            (errors/throw-service-errors :invalid-data
+                                         ["Update failed - duplicate URLs provided for granule update"]))
```
Comment (mitchstartzel) [inline ingest-app/src/cmr/ingest/services/granule_bulk_update/mime_type/echo10.clj:113]: A failure case for "listed URL not present in granule" would be a nice-to-have

File: ingest-app/src/cmr/ingest/services/granule_bulk_update_service.clj [hunk 2]
```diff
@@ -72,7 +73,7 @@
 
 (defmulti update->instruction
   "Returns the granule bulk update instruction for a single update item"
-  (fn [event-type item]
+  (fn [event-type _item]
```
Comment (mitchstartzel) [inline ingest-app/src/cmr/ingest/services/granule_bulk_update_service.clj:76]: whats the significance here?

**Review by mitchstartzel — APPROVED** [review]
Looks good, few small requests.

I agree with splitting the tests to a different file, will follow that precedent in any future BGU tickets.

File: ingest-app/src/cmr/ingest/services/granule_bulk_update_service.clj (hunk 2 above)
Comment (jaybarra) [reply ingest-app/src/cmr/ingest/services/granule_bulk_update_service.clj:76]: Clojure style-guide preferred method for marking an input as not used, alternatively just an underscore.

Comment (mitchstartzel) [reply ingest-app/src/cmr/ingest/services/granule_bulk_update_service.clj:76]: Ahh, I see. Good to know, will use this in the future.

**Review by ygliuvt — APPROVED** [review]

**Pull request merged**